lib/rules/no-unsupported-node-api-in-ssrable-components.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            description: 'Disallow unsupported Node API calls in SSR-able components.',
+            category: 'LWC',
+            url: docUrl('no-unsupported-node-api-in-ssrable-components'),
+            recommended: true,
+        },
+        messages: {
+            unsupportedNodeApi:
+                'The unsupported Node API calls are not allowed in SSR-able components.',
+        },
+        schema: [],
+    },
+    create(context) {
+        const containsunsupportedApiCall = (body) =>
+            /require|fs|child_process|worker_threads|perf_hooks/.test(body);
```

Comment:
`fs`, `child_process`, `worker_threads` and `perf_hooks` can all be imported with a `node:` prefix, so we should account for those too. See https://nodejs.org/api/fs.html#file-system

```js
import { access, constants } from 'node:fs';
```